### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by stemaMSFT**
Grammar fixes and typos but also extensive information to help users who are referring to the README for all their needs. Will update with the documentation link once docs are uploaded.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,11 +4,11 @@ A tool to bring your existing Azure resources under the management of Terraform.
 
 ## Goal
 
-`aztfexport` imports the resources that are supported by the [Terraform AzureRM provider](https://github.com/hashicorp/terraform-provider-azurerm) into the Terraform state, and generates the corresponding Terraform configuration. Both the Terraform state and configuration are expected to be consistent with the resources' remote state, i.e., `terraform plan` shows no diff. The user then is able to use Terraform to manage these resources.
+Azure Export for Terraform utilizes the [Terraform AzureRM provider](https://github.com/hashicorp/terraform-provider-azurerm) to export into Terraform state and generate the corresponding Terraform configuration. Both the Terraform state and configuration are expected to be consistent with the resources' remote state, i.e., `terraform plan` shows no diff. The user then is able to use Terraform to manage these resources.
```
Comment (magodo) [inline README.md:7]: The "export to state" sounds counter-intuitive for terraform users?

File: README.md [hunk 2]
```diff
@@ -4,11 +4,11 @@ A tool to bring your existing Azure resources under the management of Terraform.
...
 
 ## Non Goal
 
-The Terraform configurations generated by `aztfexport` are not meant to be comprehensive and do not ensure that the infrastructure can be fully reproduced from the generated configurations. For details, please refer to the [limitation](#limitation).
+The Terraform configurations generated by `aztfexport` are not meant to be comprehensive and do not ensure that the infrastructure can be fully reproduced from said generated configurations. For details, please refer to the [limitation section](#limitation).
```
Comment (magodo) [inline README.md:11]:
```suggestion
The Terraform configurations generated by `aztfexport` are not meant to be comprehensive and do not ensure that the infrastructure can be fully reproduced from said generated configurations. For details, please refer to the [limitation](#limitation).
```

I don't think the "section" here is necessary.

File: README.md [hunk 3]
```diff
@@ -88,47 +88,56 @@ Supported versions:
     apt-get install aztfexport
     ```
 
-## Precondition
+#### AUR (Linux)
 
-`aztfexport` requires a `terraform` executable installed in the `$PATH`, whose version `>= v0.12`.
+```bash
+yay -S aztfexport
+```
```
Comment (magodo) [inline README.md:95]: This is deleted as we don't officially support AUR

File: README.md [hunk 4]
```diff
@@ -88,47 +88,56 @@ Supported versions:
...
+```terminal
+aztfexport [command] [option] <scope>
+```
+Azure Export supports three variations of this core usage covered in these next sectionss: 
+- `aztfexport resource <resource id>` to import a single resource
+- `aztfexport resource-group <resource group name>` to import a resource group including child resources
+- `aztfexport query <arg where predicate>` to import a customized set of resources by an Azure Resource Graph query
 
-E.g.
+> ❗ Because each command has unique option flags available to it, `aztfexport --help` will **only list the available commands for the tool** and no option flags. **To see option flags for a given command**, please run `aztfexport [command] --help` instead.
```
Comment (magodo) [inline README.md:114]: Can we delete this as it is a universal convention?

File: README.md [hunk 5]
```diff
@@ -88,47 +88,56 @@ Supported versions:
...
 
-`aztfexport resource [option] <resource id>` exports a single resource by its Azure control plane ID.
+Then run Azure Export for Terraform in the following format:
+```terminal
+aztfexport [command] [option] <scope>
+```
+Azure Export supports three variations of this core usage covered in these next sectionss: 
+- `aztfexport resource <resource id>` to import a single resource
+- `aztfexport resource-group <resource group name>` to import a resource group including child resources
+- `aztfexport query <arg where predicate>` to import a customized set of resources by an Azure Resource Graph query
```
Comment (magodo) [inline README.md:112]: Can we simplify this a bit as the command lines are all duplicated in their own section?

File: README.md [hunk 6]
```diff
@@ -88,47 +88,56 @@ Supported versions:
...
 
-To workaround above, you can query with a bigger scope (e.g. `type =~ "microsoft.network/virtualnetworks"`) in interactive mode, then manually remove the resources other than subnets.
+`aztfexport` depends on `azlist`, which uses ARG behind the scenes. `azlist` will first make an ARG call with the given where predicate, then if `--recursive` is specified, it will recursively call "LIST" on the known child resource types. To read more about Azure Resource Graph visit the [ARG Documentation](https://learn.microsoft.com/en-us/azure/governance/resource-graph/).  
+As of now ARG only returns ARM tracked resources, but not RP proxy resources (e.g. subnet, network security rules, storage containers, etc). Thus, if you run query mode on proxy resources (e.g. `aztfexport query type =~ "microsoft.network/virtualnetworks/subnets"`), it will return nothing since subnet is not an ARM tracked resource. A workaround is to query with a bigger scope (e.g. `type =~ "microsoft.network/virtualnetworks"`) in interactive mode, then manually remove the resources other than subnets.
```
Comment (magodo) [inline README.md:136]:
```suggestion

As of now ARG only returns ARM tracked resources, but not RP proxy resources (e.g. subnet, network security rules, storage containers, etc). Thus, if you run query mode on proxy resources (e.g. `aztfexport query "type =~ 'microsoft.network/virtualnetworks/subnets'"`), it will return nothing since subnet is not an ARM tracked resource. A workaround is to query with a bigger scope (e.g. `type =~ 'microsoft.network/virtualnetworks'`) in interactive mode, then manually remove the resources other than subnets.
```

File: README.md [hunk 7]
```diff
@@ -166,48 +175,87 @@ Example:
...
+	}
 }
 ```
 
-You can generate the mapping file in all other modes (i.e. `resource`, `resource-group`, `query`) by specifying the `--generate-mapping-file` option when running non-interactively, or press <kbd>s</kbd> when running interactively in the resource list stage. Also, each run of `aztfexport` will generate the resource mapping file for you, to record what resources have been imported.
+If your use cases require pre-export modifications, you are not only welcome but also encouraged to manually construct or edit the mapping file. You could want to modify the mapping file if you:
+- Have many resources in a resource group but only wish to export a select few
+- Wish to rename all your resources in a compliant manner
+- Want to manually discover resources under a given resource type (i.e. networking or compute)
+- And other workflows you might discover!
```
Comment (magodo) [inline README.md:186]:
```suggestion
- Other workflows you might discover
```

File: README.md [hunk 8]
```diff
@@ -166,48 +175,87 @@ Example:
...
+### HCL Only
+
+For any of the aforementioned modes (`resource`, `resource-group`, `query`, `mapping-file`), users can add the `--hcl-only` flag:
+```terminal
+aztfexport [command] --hcl-only [other options] <scope>
+```
+Though `aztfexport` will by default export a state file, the `--hcl-only` flag will result in only the following being generated:
+- Any generated `.tf` HCL files
+- (Planned for 0.10, not true as of 0.9) The mapping file `.json`
+- (Planned for 0.10, not true as of 0.9) Any skipped resources in a `.txt`
```
Comment (magodo) [inline README.md:199]: Can we name the exact file names?

File: README.md [hunk 9]
```diff
@@ -166,48 +175,87 @@ Example:
...
+
+This helps for scenarios where you may not need the state or are not sure you want to generate the state. If you then wish to export all of the generated configuration to state, you can rerun the tool and utilize `aztfexport mapping-file` to do so.  
+> 💡 `--hcl-only` must target an empty directory to avoid making unwanted changes to preexisting state during the export stage (thus it will not work with `--append`, but will work with `--overwrite`). Use the `-o` flag to specify an empty directory, or if the directory does not exist, it will be created for you.
```
Comment (magodo) [inline README.md:202]:
```suggestion
> 💡 `--hcl-only` must target an empty directory to avoid making unwanted changes to preexisting state during the export stage (thus it will not work with `--append`, but will work with `--overwrite`).
```
Is this over wordy?

File: README.md [hunk 10]
```diff
@@ -166,48 +175,87 @@ Example:
...
+- <kbd>G</kbd><kbd>End</kbd> jumps to the end of the resource list
+- <kbd>/</kbd> allows you to define/apply a filter by text on the resource list
+- <kbd>Esc</kbd> clears any current filter
+- <kbd>Delete</kbd> skips the currently highlighted item and it will not be exported when <kbd>w</kbd> is pressed. If you skip a resource on accident, press <kbd>Delete</kbd> again to recover it.
+- <kbd>s</kbd> saves a [mapping file](###mapping-file) of the resource list. This file's generated list is not affected by filtering, but is affected by skipping.
+- <kbd>w</kbd> exports resources to state (if `--hcl-only` is not selected) and generates the config
+- <kbd>r</kbd> shows possible recommendations for a resource
+- <kbd>e</kbd> shows errors (if any) on exporting a resource
+- <kbd>q</kbd> quits out of interactive mode
+- <kbd>?</kbd> opens help
```
Comment (magodo) [inline README.md:227]: Do we have to be that wordy or are these not clear?

File: README.md [hunk 11]
```diff
@@ -166,48 +175,87 @@ Example:
...
+For example, to export to an [`azurerm` backend](https://www.terraform.io/language/settings/backends/azurerm#azurerm) (in this case, an Azure storage account):
 
 ```shell
 aztfexport [subcommand] --backend-type=azurerm --backend-config=resource_group_name=<resource group name> --backend-config=storage_account_name=<account name> --backend-config=container_name=<container name> --backend-config=key=terraform.tfstate 
 ```
+> 💡 Note that if the backend state already exists, `aztfexport` will merge the new resources to the existing state automatically. You do not need to specify the `--append` option.
 
-### Import Into Existing Local State
+You can also bring your own customized `provider.tf` or `terraform.tf` file with a predefined `terraform` or `provider` block; as of `v0.10` you are required to use the `--append` flag, but with the next release we hope to remove the need to use the flag to export to a backend. Be aware that your provider version should match the `aztfexport` version of AzureRM when exporting or the command will fail.
```
Comment (magodo) [inline README.md:250]: This is currently a unfixed bug, we don't want to expose this to users.

File: README.md [hunk 12]
```diff
@@ -166,48 +175,87 @@ Example:
...
 
-For local backend, `aztfexport` will by default ensure the output directory is empty at the very begining. This is to avoid any conflicts happen for existing user files, including the terraform configuration, provider configuration, the state file, etc. As a result, `aztfexport` generates a pretty new workspace for users.
+### Export Into Existing Local State
 
-One limitation of doing so is users can't import resources to existing state file via `aztfexport`. To support this scenario, you can use the `--append` option. This option will make `aztfexport` skip the empty guarantee for the output directory. If the output directory is empty, then it has no effect. Otherwise, it will ensure the provider setting (create a file for it if not exists). Then it proceeds the following steps.
+When exporting to a backend, `aztfexport` will by default ensure the output directory is empty. This is to avoid any conflicts happen for existing user files, including the terraform configuration, provider configuration, the state file, etc. As a result, `aztfexport` generates a new workspace for users.
```
Comment (magodo) [inline README.md:254]: Why do we talk about remote backend in a section that is about local state?

File: README.md [hunk 13]
```diff
@@ -217,30 +265,31 @@ This means if the output directory has an active Terraform workspace, i.e. there
 - `set`: Set a config item
 - `show`: Show the full configuration
 
-Currently, following config items are supported:
+Currently, the following config items are supported:
 
 - `installation_id`: A UUID created on first run. If there is Azure CLI or Azure Powershell installed on the current machine, the UUID will be the same value among these tools. Otherwise, a new one will be created. This is used as an identifier in the telemetry trace.
-- `telemetry_enabled`: Whether to enable telemetry? We use telemetry to identify issues and areas for improvement, in order to optimize this tool for better performance, reliability, and user experience.
+- `telemetry_enabled`: Enables telemetry. We use telemetry to identify issues and areas for improvement, in order to optimize this tool for better performance, reliability, and user experience. If you wish to disable our telemetry, set this to false.
```
Comment (magodo) [inline README.md:271]:
```suggestion
- `telemetry_enabled`: Whether to enable telemetry? We use telemetry to identify issues and areas for improvement, in order to optimize this tool for better performance, reliability, and user experience.
```

File: README.md [hunk 14]
```diff
@@ -217,30 +265,31 @@ This means if the output directory has an active Terraform workspace, i.e. there
...
 
 ## Demo
 
 [![asciicast](https://asciinema.org/a/sKYqzSiE5bpBJCB4BM2HjvF4j.svg)](https://asciinema.org/a/sKYqzSiE5bpBJCB4BM2HjvF4j)
 
-## Limitation
+## Limitations
 
-There are several limitations causing `aztfexport` can hardly generate reproducible Terraform configurations.
+We do not guarantee that redeploying Terraform configurations generated by `aztfexport` will reproduce the exact same infrastructure. This is due to known limitations that we detail below along with subsequently recommended actions. We will do our best to update this section as we discover new limitations.
```
Comment (magodo) [inline README.md:283]: This also sound unnecessarily wordy to me

File: README.md [hunk 15]
```diff
@@ -217,30 +265,31 @@ This means if the output directory has an active Terraform workspace, i.e. there
...
+One reason is because there are flexible cross-property constraints defined in the AzureRM Terraform provider. E.g. `property_a` conflicts with `property_b`. This might due to the nature of the API, or might be due to some deprecation process of the provider (e.g. `property_a` is deprecated in favor of `property_b`, but kept for backwards compatibility). These constraints require some properties to be absent in the Terraform configuration; otherwise, the configuration is not valid and will fail during `terraform validate`. For an example refer to [this schema for the azurerm_storage_blob resource](https://github.com/hashicorp/terraform-provider-azurerm/blob/8c7510d8b6ccf5098c7f05bd8c013a0e579a2e80/internal/services/storage/storage_blob_resource.go#L107-L126). If a user wishes to attempt to mitigate this issue, they would ultimately need to refer to the AzureRM documentation to know what, if any, properties are ommitted and what the current property value would entail.
```
Comment (magodo) [inline README.md:289]:
```suggestion
One reason is because there are flexible cross-property constraints defined in the AzureRM Terraform provider. E.g. `property_a` conflicts with `property_b`. This might due to the nature of the API, or might be due to some deprecation process of the provider (e.g. `property_a` is deprecated in favor of `property_b`, but kept for backwards compatibility). These constraints require some properties to be absent in the Terraform configuration; otherwise, the configuration is not valid and will fail during `terraform validate`.
```

The example is not appropriate here as we have some special handling to these constraints in the `tfadd`, where we don't want to expose the details here.

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I've taken a look through and left some comments inline. Overall, I think the readme now becomes too wordy to me. I think we can keep it short and precise, otherwise, it is both not easy to follow, and also a nontrivial effort to maintain it.

File: README.md (hunk 4 above)
Comment (stemaMSFT) [reply README.md:114]: Lots of customer feedback during the customer calls was that they were struggling to find the option flags, so I would like to make sure we keep this section.
